### PR TITLE
Make a comparison in the test more accurately

### PR DESCRIPTION
**Pull request by 777genius**
Before that, if the month coincided, then the test passed.

An example of an incorrect code, but tests passed:
```
const magic = () => new Date("2022-03-03T21:18:23.040Z")
```

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
I agree that the example above should not pass.  However, if the dates are compared at the millisecond level, there's a small risk that the tests will randomly fail.  The two dates will not necessarily be created at the same millisecond (or the same month, for that matter, but that's incredibly rare).

I think it would be better to check that `magic` contains `new Date()` with a regex.

**Comment by RandellDawson** [comment]
@iliyaZelenko Can you respond to @ojeytonwilliams's suggestion?

**Comment by iliyaZelenko** [comment]
@ojeytonwilliams You are right that this method may sometimes not work correctly.



I reset the hours, minutes, seconds, milliseconds for better accuracy using `setHours`, I was inspired by [this answer](https://stackoverflow.com/questions/492994/compare-two-dates-with-javascript#comment10495421_493018).

Now not accuracy is minimized and the date will be checked by year, month and day.
  
Maybe I missed something? Is it better to do it through regexp?

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions.english.md [hunk 1]
```diff
@@ -27,15 +27,15 @@ Rewrite the function assigned to the variable <code>magic</code> which returns a
 ```yml
 tests:
   - text: User did replace <code>var</code> keyword.
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions.english.md]:
```suggestion
   - text: User should replace <code>var</code> keyword.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions.english.md [hunk 2]
```diff
@@ -27,15 +27,15 @@ Rewrite the function assigned to the variable <code>magic</code> which returns a
...
-    testString: getUserInput => assert(!getUserInput('index').match(/var/g), 'User did replace <code>var</code> keyword.');
+    testString: getUserInput => assert(!getUserInput('index').match(/var/g));
   - text: <code>magic</code> should be a constant variable (by using <code>const</code>).
-    testString: getUserInput => assert(getUserInput('index').match(/const\s+magic/g), '<code>magic</code> should be a constant variable (by using <code>const</code>).');
+    testString: getUserInput => assert(getUserInput('index').match(/const\s+magic/g));
   - text: <code>magic</code> is a <code>function</code>.
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions.english.md]:
```suggestion
   - text: <code>magic</code> should be a <code>function</code>.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions.english.md [hunk 3]
```diff
@@ -27,15 +27,15 @@ Rewrite the function assigned to the variable <code>magic</code> which returns a
...
-    testString: assert(typeof magic === 'function', '<code>magic</code> is a <code>function</code>.');
+    testString: assert(typeof magic === 'function');
   - text: <code>magic()</code> returns correct date.
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions.english.md]:
```suggestion
   - text: <code>magic()</code> should return correct date.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions.english.md [hunk 4]
```diff
@@ -27,15 +27,15 @@ Rewrite the function assigned to the variable <code>magic</code> which returns a
...
-    testString: assert(magic().getTime() === new Date().getTime(), '<code>magic()</code> returns correct date.');
+    testString: assert(magic().setHours(0,0,0,0) === new Date().setHours(0,0,0,0));
   - text: <code>function</code> keyword was not used.
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/use-arrow-functions-to-write-concise-anonymous-functions.english.md]:
```suggestion
   - text: <code>function</code> keyword should not be used.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
Since we are working on this challenge anyway, let's fix the assert messages to make them more like all the rest in the curriculum (i.e. using `should`).

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
@ojeytonwilliams Are you OK with the changes made based on my suggestions?  If so, feel free to merge this PR.

**Comment by ojeytonwilliams** [comment]
@iliyaZelenko Thanks for the contribution! Apologies for the delay in getting it merged.

**Pull request merged**